src/features/swap/components/SwapUIMachineProvider.tsx

```diff
@@ -63,13 +63,17 @@ export function SwapUIMachineProvider({
   signMessage,
 }: SwapUIMachineProviderProps) {
   const { setValue, resetField } = useFormContext<SwapFormValues>()
+  const tokenIn = initialTokenIn || tokenList[0]
+  const tokenOut = initialTokenOut || tokenList[1]
+  if (!tokenIn || !tokenOut)
+    throw new Error("Token list must have at least 2 tokens")
```

Comment:
I think the AND isn't needed here since we want both tokenIn and tokenOut to be mandatory. Using the assert() func shouldn't cause any TS issue.
```
const tokenIn = initialTokenIn
const tokenOut = initialTokenOut
assert(tokenIn && tokenOut, "Token list must have at least 2 tokens")
```

or it could be more verbose:

```
assert(
    tokenIn !== undefined && tokenOut !== undefined,
    "Token list must have at least 2 tokens"
  )
```

Reply:
Adjusted